stocking/src/test/scala/dev/atedeg/mdm/stocking/Tests.scala

```diff
@@ -0,0 +1,71 @@
+package dev.atedeg.mdm.stocking
+
+import java.util.UUID
+
+import cats.data.{ Writer, WriterT }
+import org.scalatest.GivenWhenThen
+import org.scalatest.featurespec.AnyFeatureSpec
+import org.scalatest.matchers.should.Matchers
+
+import dev.atedeg.mdm.utils.*
+import dev.atedeg.mdm.utils.monads.*
+
+trait Mocks {
+  val batchID: BatchID = BatchID(UUID.randomUUID())
+  val cheeseType: CheeseType = 0
+  val readyForQA: Batch.ReadyForQualityAssurance = Batch.ReadyForQualityAssurance(batchID, cheeseType)
+}
+
+class Tests extends AnyFeatureSpec with GivenWhenThen with Matchers with Mocks {
+
+  Feature("Quality assurance") {
+    Scenario("An operator marks a batch as passing quality assurance") {
+      Given("a ready-for-QA batch")
+      When("the operator passes the batch")
+      val passed = approveBatch(readyForQA)
+      Then("the batch should be marked as QA-passed")
+      passed shouldBe a[QualityAssuredBatch.Passed]
+      passed.id shouldEqual batchID
+      passed.cheeseType shouldEqual cheeseType
+    }
+    Scenario("An operator marks a batch as failing quality assurance") {
+      Given("a ready-for-QA batch")
+      When("the operator fails the batch")
+      val failed = rejectBatch(readyForQA)
+      Then("the batch should be marked as QA-failed")
+      failed shouldBe a[QualityAssuredBatch.Failed]
+      failed.id shouldEqual batchID
+      failed.cheeseType shouldEqual cheeseType
+    }
+    Scenario("An operator tries to print a label for a cheese within weight range from a batch") {
+      Given("a batch")
+      val passed = approveBatch(readyForQA)
+      val cheeseTypeWeight = WeightInGrams(1)
+      When("the operator prints a label for a product within weight range")
+      val labelledMonad: Action[WeightNotInRange, OutgoingEvent.ProductStocked, LabelledProduct] =
+        labelProduct(passed, cheeseTypeWeight)
```

Comment:
You could import `OutgoingEvent.*` and rename `cheeseTypeWeight` to the shorter `weight` and this would fit in one line:
```suggestion
      val labelledMonad: Action[WeightNotInRange, ProductStocked, LabelledProduct] = labelProduct(passed, weight)
```